src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClient.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Execution;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Remote
+{
+    /// <summary>
+    /// This lets users create a session to communicate with remote host (i.e. ServiceHub)
+    /// </summary>
+    internal abstract class RemoteHostClient
```

Comment:
>RemoteHostClient [](start = 28, length = 16)

I think we can simplify some of this, but this is OK and it'll probably be much easier to simplify this once it's in master rather than doing it through the review.